ckanext/switzerland/dcat/harvesters.py

```diff
@@ -131,3 +132,35 @@ def before_create(self, harvest_object, dataset_dict, temp_dict):
         except ValueError:
             # nothing configured
             pass
+
+    def before_update(self, harvest_object, dataset_dict, temp_dict):
+        # get existing pkg_dict with incoming pkg_name
+        site_user = logic.get_action('get_site_user')(
+            {'model': model, 'ignore_auth': True}, {})
+        context = {
+            'model': model,
+            'session': model.Session,
+            'ignore_auth': True,
+            'user': site_user['name'],
+        }
+        existing_pkg = p.toolkit.get_action('package_show')(context, {
+            'id': dataset_dict.get('name')})
+
+        # get existing resource-identifiers
+        existing_resources = existing_pkg.get('resources')
+        existing_resource_ids = []
+        for existing_resource in existing_resources:
+            if existing_resource.get('identifier'):
+                existing_resource_ids.append(existing_resource.get('identifier'))  # noqa
```

Comment:
I was think again about this part. And actually here we only want to generate a list (or even better a set) of identifiers that are in the existing package. Nothing else. I think we can write this in one line:

```python
identifiers = set([r.get('identifier') for r in existing_pkg.get('resources')])
```